docs: sequence controller w/ clearer descriptions

stijnmoreels opened the pull request:
A proposal for a more dev/user-friendly description on the **Sequence Controller** by first going over the motivation and reason, and afterwards laying more the focus on the 'What' the component does instead of 'How' it does it.

stijnmoreels commented:
@pim-simons , can you also give your opinion on this? Especially on the 'pseudo Logic App workflow' and the motivations. Hopefully it gets more clear for all of us. 😉
Thx in advance!

GoutsmitSam reviewed (commented):
@stijnmoreels  The internal links to the different sequencecontroller actions don't work for me in the Deploy Preview page (for example https://deploy-preview-230--invictus-for-azure.netlify.app/#/framework/components/sequencecontroller?id=2-wait-for-sequence)

Apart from that, it seems nice and clear to me . Thanks!

stijnmoreels commented:
> @stijnmoreels The internal links to the different sequencecontroller actions don't work for me in the Deploy Preview page (for example https://deploy-preview-230--invictus-for-azure.netlify.app/#/framework/components/sequencecontroller?id=2-wait-for-sequence)
> 
> Apart from that, it seems nice and clear to me . Thanks!

Fixed that by a heading underscore `_` as the headers start with a number (and Docusaurus didn't picked this up). Fixed this here: https://github.com/invictus-integration/docs-ifa/pull/230/commits/983960d4317dada7493fba060db4cde4fe6735ec

pim-simons commented on framework/components/sequencecontroller.md:
```diff
@@ -1,52 +1,115 @@
+## Motivation
 
-The Sequence controller will be mainly used to process LogicApp runs in order. The Sequence Controller exposes 3 actions:
+Some dependent external systems can't handle parallel message processing or requires a certain order of messages. Sending messages serially can be a solution, at the cost of a big performance impact. If the processing or sending of a single message is too big, serial is still not the solution.
```
```suggestion
Some dependent external systems can't handle parallel message processing or requires a certain order of messages. Sending messages serially (for example by setting the concurrency on a Logic App to `1`) can be a solution, at the cost of a big performance impact. If the processing or sending of a single message is too big, serial is still not the solution.
```

pim-simons commented on framework/components/sequencecontroller.md:
```diff
@@ -1,52 +1,115 @@
+
+* [Get sequence number](#_1-get-sequence-number): allows the workflow to determine what current position it has in the sequence;
+* [Wait for sequence](#_2-wait-for-sequence): allows the workflow to wait its turn, doing the actual work in a **Control** task;
+* [Complete sequence](#3-complete-sequence): allows the workflow to signal that the next workflow in the sequence is up.
```
```suggestion
* [Complete sequence](#_3-complete-sequence): allows the workflow to signal that the next workflow in the sequence is up.
```

this link doesn't work yet 😉

pim-simons commented on framework/components/sequencecontroller.md:
```diff
@@ -1,52 +1,115 @@
-### WaitForExecution
+### 2. Wait for sequence
+
+The next step for the Logic App workflow to run in sequence, is to wait its turn. To facilitate this, the counter collected from the previous [Get sequence number](#1-get-sequence-number) step is required.
```
```suggestion
The next step for the Logic App workflow to run in sequence, is to wait its turn. To facilitate this, the counter collected from the previous [Get sequence number](#_1-get-sequence-number) step is required.
```

pim-simons commented on framework/components/sequencecontroller.md:
```diff
@@ -1,52 +1,115 @@
+## Motivation
 
-The Sequence controller will be mainly used to process LogicApp runs in order. The Sequence Controller exposes 3 actions:
+Some dependent external systems can't handle parallel message processing or requires a certain order of messages. Sending messages serially can be a solution, at the cost of a big performance impact. If the processing or sending of a single message is too big, serial is still not the solution.
```
I don't quite understand this line

> If the processing or sending of a single message is too big, serial is still not the solution.


I think a better example to give is the following real-life example from a customer.
A downstream application cannot handle multiple calls at the same time for the same orderid, by using the sequencecontroller we are able to make sure that a single call per orderid is done (by using the orderid in the sequencename) and we do not have the use the concurrency option for the Logic App which would dramatically slow down the processing for all messages.

stijnmoreels replied:
Thx a lot for this, Pim!
Added 'a story from the trenches' here: https://github.com/invictus-integration/docs-ifa/pull/230/commits/6b80f615e9f6d332d86998533ad6953f7839c06b

pim-simons reviewed (changes requested):
Some small comments, but looks good overall 🥇

stijnmoreels commented on framework/components/sequencecontroller.md:
```diff
@@ -1,52 +1,119 @@
+
+> 💡 This step can be circumvented in some advanced scenarios where you keep track of the order numbers and pass it yourself during the [Wait for sequence](#_2-wait-for-sequence).
+
+The most simple request body only contains the name of the sequence. This name can be seen as the transactional group that chain all related workflows. Assuming this is the first workflow, it will receive number 1 in the response body; the second workflow will receive number 2, and so forth.
```
When no `sequenceStart` is defined, it currently returns `0` as the 'first' number, but there's a PR that adds `1` as the new default.

stijnmoreels replied:
Discussed separately: it is not required to go tht much in detail.
